src/EditorFeatures/Core/InheritanceMargin/AbstractInheritanceMarginService.cs

```diff
@@ -84,48 +89,49 @@ internal abstract partial class AbstractInheritanceMarginService : IInheritanceM
 
         private async Task AddInheritanceMemberItemsForNamedTypeAsync(
             Solution solution,
-            SourceText sourceText,
+            TextLineCollection lines,
             SyntaxNode declarationNode,
             INamedTypeSymbol memberSymbol,
-            ArrayBuilder<InheritanceMemberItem> builder,
+            ArrayBuilder<InheritanceMarginItem> builder,
             CancellationToken cancellationToken)
         {
             // Get all base types.
-            var allBaseTypes = BaseTypeFinder.FindBaseTypesAndInterfaces(memberSymbol);
+            var allBaseSymbols = BaseTypeFinder.FindBaseTypesAndInterfaces(memberSymbol);
 
             // Filter out
             // 1. System.Object. (otherwise margin would be shown for all classes)
             // 2. System.ValueType. (otherwise margin would be shown for all structs)
             // 3. System.Enum. (otherwise margin would be shown for all enum)
-            var baseTypes = allBaseTypes
-                .WhereAsArray(symbol => !IsUnwantedBaseType((ITypeSymbol)symbol));
+            var baseSymbols = allBaseSymbols
+                .WhereAsArray(symbol => ((ITypeSymbol)symbol).SpecialType is not (SpecialType.System_Object or SpecialType.System_ValueType or SpecialType.System_Enum));
```

Comment:
FindBaseTypesAndInterfaces should be returning an IMmutableArray<INamedTypeSymbol>.  If not, we should fix that guy, instead of forcing these unpleasant cassts here :)